contracts/Zap.sol

```diff
@@ -183,7 +182,19 @@ contract Zap is Ownable, ReentrancyGuard {
 
         (uint256 amountTo, address lpToken) = _performZapOut(zap, swapTokenA, swapTokenB);
 
-        amountTransferred = _getFeeAndTransferTokens(tokenTo, amountTo, receiver, affiliate);
+        uint256 totalProtocolFeePortion;
+        if (tokenTo == address(0)) {
+            // unwrap to native currency
+            IWETH(nativeCurrencyWrapper).withdraw(amountTo);
+            totalProtocolFeePortion = _subtractProtocolFee(nativeCurrencyAddress, amountTo, affiliate);
+            TransferHelper.safeTransferETH(receiver, amountTo - totalProtocolFeePortion);
+        } else {
+            totalProtocolFeePortion = _subtractProtocolFee(tokenTo, amountTo, affiliate);
+            TransferHelper.safeTransfer(tokenTo, receiver, amountTo - totalProtocolFeePortion);
+        }
+
+        amountTransferred = amountTo - totalProtocolFeePortion;
```

Comment:
Why is it better to move the code and delete the function?

Reply:
Gas costs mainly, if it were being used in more places I would be in favour of readability/maintainability/less-code-repetition but it's not the case.

Reply:
Also, it's a quite short function so it doesn't affect readability/undestanding much...